### ISSUE TITLE
"EntityFrameworkEvent" missing from json when using AzureStorageBlobs data provider with EntityFramework.Core

### ISSUE THREAD

**Issue by mahgo** [thepirat000/Audit.NET#476]
**Describe the bug**
When using Audit.EntityFramework.Core to generate Audit Logs, "EntityFrameworkEvent" is missing from the json when the data provider is Audit.NET.AzureStorageBlobs. This is not an issue when using other data providers, such as File or MySQL. I have Audit.EntityFramework.Core set up via interceptors. For example:
- From AzureStorageBlobs:
```
{
    "Environment": {
        "UserName": "user",
        "MachineName": "50475ac04d0f",
        "DomainName": "50475ac04d0f",
        "CallingMethodName": "Audit.EntityFramework.DbContextHelper\u002B\u003CCreateAuditScopeAsync\u003Ed__31.MoveNext()",
        "AssemblyName": "Audit.EntityFramework.Core, Version=19.0.4.0, Culture=neutral, PublicKeyToken=571d6b80b242c87e",
        "Culture": ""
    },
    "EventType": "DataContext",
    "StartDate": "2022-01-27T23:37:07.1285766Z",
    "EndDate": "2022-01-27T23:37:07.4429839Z",
    "Duration": 314
}
```
From file provider:
```
{
    "EntityFrameworkEvent": {
        "Database": "testdatabase",
        "ContextId": "039dffcb-2fe0-415c-b188-50d2e1fefe40:0",
        "Entries": [{
                "Table": "Test",
                "Name": "Test",
                "PrimaryKey": {
                    "Id": "ea936ecc-bbff-49e8-8fdf-c11c09c7edf6"
                },
                "Action": "Insert",
                "Entity": {
                    "Name": "A",
                    "Id": "ea936ecc-bbff-49e8-8fdf-c11c09c7edf6",
                },
                "ColumnValues": {
                    "Id": "ea936ecc-bbff-49e8-8fdf-c11c09c7edf6",
                    "Name": "A",
                },
                "Valid": true
            }
        ],
        "Result": 1,
        "Success": true
    },
    "Environment": {
        "UserName": "user",
        "MachineName": "50475ac04d0f",
        "DomainName": "50475ac04d0f",
        "CallingMethodName": "Audit.EntityFramework.DbContextHelper\u002B\u003CCreateAuditScopeAsync\u003Ed__31.MoveNext()",
        "AssemblyName": "Audit.EntityFramework.Core, Version=19.0.4.0, Culture=neutral, PublicKeyToken=571d6b80b242c87e",
        "Culture": ""
    },
    "EventType": "DataContext",
    "StartDate": "2022-01-28T00:07:13.4842216Z",
    "EndDate": "2022-01-28T00:07:13.8289316Z",
    "Duration": 345
}
```

**To Reproduce**
Use Audit.EntityFramework.Core with Audit.NET.AzureStorageBlobs data provider.

**Expected behavior**
EntityFrameworkEvent should exist when using the AzureStorageBlobs data provider.

**Libraries (specify the Audit.NET extensions being used including version):**
 - Audit.EntityFramework.Core: 19.0.4
 - Audit.NET.AzureStorageBlob: 19.0.4

**Target .NET framework:**
 - .NET 6

**Comment by mahgo** [comment]
This is also an issue with the AuditCommandInterceptor json missing "CommandEvent". Maybe this is some kind of inheritance/serialization issue?

**Comment by thepirat000** [comment]
Oh, I see what the problem is.

In this line:

https://github.com/thepirat000/Audit.NET/blob/999ac7b5b2309082a34cce92e6383ba890bff421/src/Audit.NET.AzureStorageBlobs/Providers/AzureStorageBlobDataProvider.cs#L184 

The implementation of [`BinaryData.FromObjectAsJson<T>(T jsonSerializable)`](https://docs.microsoft.com/en-us/dotnet/api/system.binarydata.fromobjectasjson?view=dotnet-plat-ext-6.0#system-binarydata-fromobjectasjson-1(-0-system-text-json-jsonserializeroptions)) uses `typeof(T)` instead of `jsonSerializable.GetType()` to get the type to serialize. That's why the information not belonging to the type T is ignored.

I will change the `BinaryData.FromObjectAsJson<T>` call and use the [BinaryData constructor](https://docs.microsoft.com/en-us/dotnet/api/system.binarydata.-ctor?view=dotnet-plat-ext-6.0#system-binarydata-ctor(system-object-system-text-json-jsonserializeroptions-system-type)) from an object instead, which uses the `jsonSerializable.GetType()`

**Comment by thepirat000** [comment]
This was fixed on version 19.0.5, please upgrade your references and re-test

**Comment by mahgo** [comment]
Thanks, this fixed it.
